### PR TITLE
chore: test parallel executions

### PR DESCRIPTION
**Pull request by christyjacob4**
## What does this PR do?

Trying to parallelise Appwrite tests, extending upon the work done in PR 
https://github.com/appwrite/appwrite/pull/5877

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -34,27 +27,96 @@ jobs:
...
+        run: docker compose exec appwrite test /usr/src/code/tests/unit 
+
+  e2e_test:
+    name: E2E Test
+    runs-on: ubuntu-latest
+    needs: setup
+    strategy:
+      fail-fast: false
+      matrix:
+        services: [Account,Avatars,Console,Databases,Functions,GraphQL,Health,Locale,Projects,Realtime,Storage,Teams,Users,Webhooks]
```
Comment (Meldiron) [inline .github/workflows/tests.yml:98]: Can we rename this to `service`? That way later we can do `Services/${{matrix.service}}`, sounds more verbose than having plural there.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -34,27 +27,96 @@ jobs:
...
+
+      - name: Run Unit Tests
+        run: docker compose exec appwrite test /usr/src/code/tests/unit 
+
+  e2e_test:
+    name: E2E Test
+    runs-on: ubuntu-latest
+    needs: setup
+    strategy:
+      fail-fast: false
```
Comment (Meldiron) [inline .github/workflows/tests.yml:96]: If there is specific reason why this is here, lets add it as comment.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -34,27 +27,96 @@ jobs:
...
+    
+      - name: Run ${{matrix.services}} Tests
+        run: docker compose exec -T appwrite test /usr/src/code/tests/e2e/Services/${{matrix.services}} --debug
 
-    - name: Run Tests
-      run: docker compose exec -T appwrite test --debug
\ No newline at end of file
+      - name: Output build logs
+        if: always() && ${{ matrix.services == 'GraphQL' }}
+        run: docker compose logs appwrite-worker-builds
```
Comment (Meldiron) [inline .github/workflows/tests.yml:122]: Feels like leftover

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -34,27 +27,96 @@ jobs:
...
+      uses: actions/cache@v3
+      with:
+        key: ${{ env.IMAGE }}-${{ github.sha }}
+        path: /tmp/${{ env.IMAGE }}.tar
+
+    # - name: Upload Docker Image
+    #   uses: actions/upload-artifact@v3
+    #   with:
+    #     name: ${{ env.IMAGE }}
+    #     path: /tmp/${{ env.IMAGE }}.tar
```
Comment (Meldiron) [inline .github/workflows/tests.yml:58]: Leftover, above is same thing but with cache instead of uploading

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -34,27 +27,96 @@ jobs:
...
-      run:  |
-        docker compose up -d
-        sleep 30
+    # - name: Pull Docker Images
+    #   run: docker compose pull --ignore-pull-failures --ignore-buildable
+    
+    # - name: List Files and images
+    #   run: |
+    #     docker compose config --images | sort -u
+    #     docker save $(docker compose config --images | sort -u) -o /tmp/allinone.tar
```
Comment (Meldiron) [inline .github/workflows/tests.yml:46]: If this is relevant, lets add comment and explain a little what it is. `tmp/allinone` is confusing me.

If not relevant, lets delete

File: .github/workflows/tests.yml (hunk 1 above)
Comment (christyjacob4) [reply .github/workflows/tests.yml]: done

File: .github/workflows/tests.yml (hunk 3 above)
Comment (christyjacob4) [reply .github/workflows/tests.yml]: removed

File: .github/workflows/tests.yml (hunk 4 above)
Comment (christyjacob4) [reply .github/workflows/tests.yml]: removed

**Review by eldadfux — APPROVED** [review]

**Pull request merged**